### PR TITLE
HOSTEDCP-1460: Enable Azure conformance tests for Hypershift

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**

### CODE REVIEW

**Comment by Patryk-Stefanski** [comment]
/test all
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/test all
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/test all
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/test all
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/test all
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure
/test all

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/test all
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/test all
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/test all
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/test all
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/test all
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/test all
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure
/test all

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

File: ci-operator/config/openshift-priv/hypershift/openshift-priv-hypershift-main.yaml [hunk 1]
```diff
@@ -34,6 +34,11 @@ promotion:
     name: 4.16-priv
     namespace: ocp-private
 releases:
+  4-16-nightly:
```
Comment (bryan-cox) [inline ci-operator/config/openshift-priv/hypershift/openshift-priv-hypershift-main.yaml:37]: Should we go ahead and set this to 4.17 nightly? Those are already exist and branching is next week.

File: ci-operator/config/openshift-priv/hypershift/openshift-priv-hypershift-release-4.17.yaml [hunk 2]
```diff
@@ -34,6 +34,11 @@ promotion:
     name: 4.17-priv
     namespace: ocp-private
 releases:
+  4-16-nightly:
```
Comment (bryan-cox) [inline ci-operator/config/openshift-priv/hypershift/openshift-priv-hypershift-release-4.17.yaml:37]: This should be set to 4.17 nightly, no?

File: ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml [hunk 3]
```diff
@@ -33,6 +33,11 @@ promotion:
     name: "4.16"
     namespace: ocp
 releases:
+  4-16-nightly:
```
Comment (bryan-cox) [inline ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml:36]: Similar comment as above.

File: ci-operator/config/openshift/hypershift/openshift-hypershift-release-4.17.yaml [hunk 4]
```diff
@@ -33,6 +33,11 @@ promotion:
     name: "4.17"
     namespace: ocp
 releases:
+  4-16-nightly:
```
Comment (bryan-cox) [inline ci-operator/config/openshift/hypershift/openshift-hypershift-release-4.17.yaml:36]: Similar comment as above.

**Review by bryan-cox — COMMENTED** [review]
Just a few questions

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by LiangquanLi930** [comment]
@Patryk-Stefanski To be honest, I don't want to merge `hypershift-azure-create` and `hypershift-aws-create` into `hypershift-create`, for the following reasons:
1. Merging them would create a huge step that would be difficult to maintain.
2. If we fully follow this change, it means we also need to adjust kubevirt and agent.
3. The design of prow should be modular, rather than adding a lot of if-else statements.

**Comment by Patryk-Stefanski** [comment]
@LiangquanLi930 We already use this approach when we create our nested mgmt clusters and by doing this I think it makes it easier to maintain if anything as its one file rather than one for each environment.

With that being said I don't mind either or, @sjenning are you okay with having separate files for create/delete for each env?

**Comment by LiangquanLi930** [comment]
> @LiangquanLi930 We already use this approach when we create our nested mgmt clusters and by doing this I think it makes it easier to maintain if anything as its one file rather than one for each environment.
> 
> With that being said I don't mind either or, @sjenning are you okay with having separate files for create/delete for each env?

Thank you both for sharing your perspectives. I understand the benefits of having a single file for creating nested management clusters, but I still have reservations about merging hypershift-azure-create and hypershift-aws-create into hypershift-create.

Maintaining a single, large file could become challenging over time, especially if it grows significantly due to the addition of multiple environments. It might also introduce complexities in managing changes and debugging issues.

Additionally, if we proceed with this merger, it would require adjustments not only in hypershift-create but also in kubevirt and agent, which could increase the scope of work and potentially introduce more points of failure.

Instead of adding a lot of if-else statements, I believe it's essential to keep the design of prow modular to ensure flexibility and ease of maintenance. By maintaining separate files for creating and deleting resources for each environment, we can adhere to this modular approach while still keeping our codebase manageable and scalable.

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
pull-ci-openshift-hypershift-release-4.16-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.16-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.16-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.16-e2e-conformance-azure

**Comment by bryan-cox** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.16-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.16-e2e-conformance-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.16-e2e-conformance-azure

**Pull request merged**